internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubSubMessage.java

```diff
@@ -35,6 +35,10 @@ default String getTopicName() {
     return getTopicPartition().getPubSubTopic().getName();
   }
 
+  default PubSubTopic getTopic() {
+    return getTopicPartition().getPubSubTopic();
+  }
+
```

Comment:
Is this really needed? Looks like there are few methods that already seem redundant without any concrete need. e.g., 

`getTopicName` and `getTopic`; ideally if have a way to get topic information everything about the topic should be abstracted and shouldn't be exposed as another getter.

Similar to pattern is also followed for partition where we have `getTopicPartition` and `getPartition` which are both operating on the object TopicPartition. 

I don't see a benefit why callers can't do this indirection directly as opposed exposing it through a getter (didn't notice any concrete need to be overridden or implemented by the implementors of the interface).

Reply:
Let's not confuse leaky abstractions with convenience methods. 

- `getTopicName` returns the name of the topic as a `String`.
- `getTopic` returns an instance of type `PubSubTopic`.

The same applies for `PubSubTopicPartition`.

> Is this really needed?

No but it helps in reducing verbose pattern: `record.getTopicPartition().getPubSubTopic()`  to `record.getTopic()` without any downsides. 

_That being said, I do think both `PubSubTopic` and `PubSubTopicPartition` abstractions are unnecessary. Just using `String` and `int` would be sufficient_

Reply:
>Let's not confuse leaky abstractions with convenience methods.
I differ here. I don't see these as convenience methods. As far as this interface is considered these are seen as contracts. If at all they are convenience methods, they shouldn't part of an interface as the implementors can choose to evolve and override and cause changes in assumptions of these representations.

It is not in the nature of this interface to hold convenience methods and utility classes are more of a fit. Typically utility classes are final and non extensible enforcing one and only behavior of the method thereby making it easy to rationalize the behavior as opposed overrides making it more complex to identify and reason about the behavior and its whereabouts (especially when its not needed).

Additionally, `TopicPartition` and `PubSubTopic` are the contracts this interface works with and exposes. It is not the inner contracts of how partition is represented within `TopicPartition` and how topic is represented as a string within `PubSubTopic`. I don't see why this interface should be responsible for handling evolutions within its contracts `TopicPartition` and `PubSubTopic`.

Reply:
>they shouldn't part of an interface as the implementors can choose to evolve and override and cause changes in assumptions of these representations.

Even if this is the case, API by contract needs to return an instance of type `PubSubTopic`.

Reply:
I believe this can be simplified as follows:

* Each message belongs to a topic.
* Each message is assigned to a specific partition within that topic.
* This info about messages should be easily retrievable from the message itself.

Reply:
we had a sync offline about this. This was more of an observation and comment on the pattern and lay of the land of the usage in the current class. However, I understand its not within the scope of PR to necessarily act on any of the above discussions.

For the context of the PR, feel free to treat it as a non-blocker conversation and merge it.